### ISSUE TITLE
Request: Show rustdoc warnings

### ISSUE THREAD

**Issue by KSXGitHub** [rust-lang/rust-analyzer#9131]
Example code:

```rust
use super::NewName;

/// Create an instance of [`OldName`].
pub fn create() -> NewName {
    unimplemented!()
}
```

In the code above, `OldName` has been renamed to `NewName` but I forget updating the doc. Running `cargo doc` would print warnings but I don't always remember it.

**Request:** Show rustdoc warnings like compiler warnings.

**Edit:** One had suggested that I set `rust-analyzer.checkOnSave.command` to `doc`, but I have already set it to `clippy`.

**Comment by bjorn3** [comment]
You can use `"rust-analyzer.checkOnSave.command": "doc"` to run `cargo doc` instead of `cargo check`. It is not possible to get all warnings in doc comments as rustc doesn't interpret doc comments beyond what is necessary to parse your code. Rustdoc is responsible for getting them. I guess it should be possible to implement a warning for unresolved item links inside doc comments directly inside rust-analyzer though. They are already resolved I believe for semantic syntax highlighting.

**Comment by KSXGitHub** [comment]
@bjorn3 I currently set it to `clippy`. Would `doc` also report Clippy lints?

**Comment by bjorn3** [comment]
No, it wouldn't. There is no way to get both rustdoc and clippy lints without running both tools.

**Comment by Walther** [comment]
Would it be possible to consider a feature request then for the `rust-analyzer.checkOnSave.command` (or a new setting) to support a list of commands then? I would guess many people would appreciate seeing all three of `rustc`, `clippy` and `doc` issues in the IDE "Problems" lists that `rust-analyzer` is able to populate - without having to switch back and forth between the commands

**Comment by lnicola** [comment]
`cargo clippy` should also `rustc` warnings. For `cargo doc`, making a shell script or batch file that calls both might work.

**Comment by Walther** [comment]
You are describing a possible workaround, which I appreciate being brought up as it could unblock some people.

However - is it completely out of question for `rust-analyzer` to consider supporting this by itself - considering how common it is to use the language's builtin documentation support?

**Comment by lnicola** [comment]
I don't know if it's out of scope. I certainly wouldn't use it because it slows down the builds and maybe I save a file 100 times, but only run `cargo doc` once (if even).

But it's totally fair to request ask for it.

**Comment by soareschen** [comment]
When I changed the `checkOnSave` command to `doc`, I get the following error:

```
[ERROR flycheck] Flycheck failed to run the following command: "cargo" "doc" "--workspace" "--message-format=json-diagnostic-rendered-ansi" "--manifest-path" "/path/to/Cargo.toml" "--all-targets" "--all-features"
[ERROR rust_analyzer::lsp_utils] cargo check failed:
Cargo watcher failed, the command produced no valid metadata (exit code: ExitStatus(unix_wait_status(256))):
error: Found argument '--all-targets' which wasn't expected, or isn't valid in this context

  Did you mean '--all-features'?

Usage: cargo doc --workspace --message-format <FMT> --manifest-path <PATH> --all-features

For more information try '--help'
```

So there is really no viable workaround to check for rustdoc warnings within Rust Analyzer.

**Comment by melody-rs** [comment]
It's possible to fix `--all-targets` error by doing this:
```jsonc
  // .vscode/settings.json
  "rust-analyzer.check.command": "doc",
  "rust-analyzer.check.allTargets": false,
```

**Comment by ChayimFriedman2** [comment]
The best you can have if you want both rustdoc and Clippy is to have some script r-a will call that will execute both and handle returning diagnostics.
